spec/imsc-hrm.html

```diff
@@ -519,7 +525,8 @@ <h2>Paint Images</h2>
       <h2>Paint Text</h2>
 
       <p>In the context of this section, a <dfn>glyph</dfn> is a tuple consisting of (i) one character and (ii) the computed values
-      of the following style properties:</p>
+      of the following style properties, excluding those properties that are associated with a Feature or Extension that is designate
+      at <em>prohibited</em> in the profile to which the <a>Document Instance</a> conforms:</p>
```

Comment:
```suggestion
      of the following style properties, excluding those properties that are associated with a Feature or Extension that is designated
      as <em>prohibited</em> in the highest version profile to which the <a>Document Instance</a> claims conformance:</p>
```
Two problems here:
1. Grammar/typo
2. A document is likely to conform to multiple versions of whichever of Text or Image Profile it is, so in that case, which version defines the properties to be excluded from the tuple?

I've suggested a resolution based on the document conformance claim, should it be present, but that's not required, so it doesn't seem ideal.

Maybe an **even better** solution would be, instead of excluding style properties that are prohibited in any profile version, use the resulting computed value based on the default for that style property, as though it were permitted. If the feature is prohibited and the document conforms to that prohibition, then all the characters will have the same computed value for that style property, and the >1st occurrence will therefore be available in the glyph copy buffer. So it should have the same result in the end, as what I think we intend here.

Reply:
What about referring to the _effective content profile_ of the IMSC Document Instance? This allows the profile against which the HRM is run to be determined both through the document processing context and through inline data, e.g. ttp:profile.

Reply:
Interesting approach. To get the effective content profile you effectively have to run a defined process derive it, which is a new requirement; also the result might not be an IMSC profile at all, so we might have to inject some state into the process to ensure that, in the absence of any other data, at least one IMSC profile is derived as the effective content profile.

I think one of the things I'm concerned about that I haven't mentioned so far is that we need a construct that allows us to manage the results of HRM processing of the same document against different versions of IMSC. E.g. a single document, considered as, say, an IMSC 1.0.1 Text profile effective content profile, and processed against the HRM, might somehow give a different result to the same document considered as an IMSC 1.2 Text profile document.

That seems to be a situation we should avoid occurring _accidentally_, but we might want to introduce it deliberately later if we want to track increased device performance, say

Aside from these considerations, I'm not sure if there's something unpalatable about my proposed alternative of just assuming a value for prohibited features?

Reply:
> I think one of the things I'm concerned about that I haven't mentioned so far is that we need a construct that allows us to manage the results of HRM processing of the same document against different versions of IMSC.

Exactly. HRM validation, just like syntactic validation, needs to happen against a defined profile of IMSC.

How the defined profile of IMSC is determined depends on the circumstances:

* usually it will be specified by the user or otherwise burned in the system
* it can also be inferred from the document, but this is an exception rather than a rule, especially in the case of IMSC, which does not require profile conformance to be signaled.

So I think it is fair to condition the HRM validation on the knowledge of an effective content profile that is one of the IMSC profiles.

Reply:
Maybe we're looking at this the wrong way up. We're defining an HRM that we expect to be referenced in future editions of IMSC. We can make it a requirement of reference that the referencing document must define the set of style attributes to be included in the tuple.